command/agent/cache/proxy.go

```diff
@@ -49,10 +49,6 @@ type Proxier interface {
 // NewSendResponse creates a new SendResponse and takes care of initializing its
 // fields properly.
 func NewSendResponse(apiResponse *api.Response, responseBody []byte) (*SendResponse, error) {
-	if apiResponse == nil {
```

Comment:
OK so you moved this here, will the assumptions always be true that we don't want to cache a nil response? Might want to check with Vishal/Calvin.

Reply:
Yes. We never want to cache a nil response.

Reply:
Can you evaluate the change more generally?  I think doing the check outside NewSendResponse is an improvement because it lets us report the error, but I don't know if there were other thoughts that went into the current design.  The above is the only caller of NewSendResponse with a non-literal first argument btw.